src/parser/jobs/nin/modules/TrickAttackUsage.tsx

```diff
@@ -0,0 +1,126 @@
+import {Trans, Plural} from '@lingui/react'
+import {ActionLink} from 'components/ui/DbLink'
+import {ActionKey} from 'data/ACTIONS'
+import {Event, Events} from 'event'
+import {Analyser} from 'parser/core/Analyser'
+import {EventHook} from 'parser/core/Dispatcher'
+import {filter} from 'parser/core/filter'
+import {dependency} from 'parser/core/Injectable'
+import {Data} from 'parser/core/modules/Data'
+import Downtime from 'parser/core/modules/Downtime'
+import Suggestions, {Suggestion, TieredSuggestion, SEVERITY} from 'parser/core/modules/Suggestions'
+import React from 'react'
+
+const OPTIMAL_GCD_COUNT = 5 // Opener should be Suiton > AE combo > SE before Trick
+
+const MUDRAS: ActionKey[] = [
+	'TEN',
+	'TEN_KASSATSU',
+	'CHI',
+	'CHI_KASSATSU',
+	'JIN',
+	'JIN_KASSATSU',
+]
+
+export class TrickAttackUsage extends Analyser {
+	static override handle = 'taUsage'
+
+	@dependency private data!: Data
+	@dependency private downtime!: Downtime
+	@dependency private suggestions!: Suggestions
+
+	private mudraActions: number[] = []
+
+	private taCasts: number[] = []
+	private lostTime: number = 0
+	private gcdCount: number = 0
+	private castHook?: EventHook<Events['action']>
+
+	override initialise() {
+		this.mudraActions = MUDRAS.map(k => this.data.actions[k].id)
+
+		const playerFilter = filter<Event>().source(this.parser.actor.id)
+		this.castHook = this.addEventHook(playerFilter.type('action'), this.onCast)
```

Comment:
A comment here explaining why it's initialised before the TA cast might be useful. I assume it's for some kind of pre-pull memes.

Reply:
No memes, this is for opener tracking purposes. There's a suggestion that dings you if your first Trick isn't weaved on the nth (currently 5th, may change in EW) GCD. I can add a comment tho.